src/OSSupport/StackTrace.cpp

```diff
@@ -34,11 +36,35 @@ void PrintStackTrace(void)
 		sw.ShowCallstack();
 	#else
 		#ifdef __GLIBC__
-			// Use the backtrace() function to get and output the stackTrace:
-			// Code adapted from https://stackoverflow.com/questions/77005/how-to-generate-a-stacktrace-when-my-gcc-c-app-crashes
-			void * stackTrace[30];
+			auto output = stderr;
+			void * stackTrace[66];
+			char * demangled = nullptr;
+			size_t demangledLen = 0;
+			int demangledStatus;
+			std::regex lineSplitter ("^([^(]+)\\(([^+]+)\\+(?:[^)]+)\\) \\[([^\\]]+)\\]$");
+
 			auto numItems = backtrace(stackTrace, ARRAYCOUNT(stackTrace));
-			backtrace_symbols_fd(stackTrace, numItems, STDERR_FILENO);
+			char** buffer = backtrace_symbols(stackTrace, numItems);
+
+			for (auto frameId = (numItems - 1); frameId > 0; frameId--)
+			{
+				bool handled = false;
+				std::cmatch m;
+				if (std::regex_match (buffer[frameId], m, lineSplitter))
```

Comment:
Do we need to match things with a regex, can't we just send everything for demangling?

I just fear that once a crash is hit, nothing can be considered stable enough, especially memory allocation.